From 323dbb8336816b37eaf08c1d0e904180c2ca94e9 Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Thu, 29 Feb 2024 15:28:46 +0100
Subject: [PATCH 0694/1695] dts: stm32: set the GPIOs used for wake-up as
 secure on stm32mp135f-dk

Set these GPIOs as secure to avoid the non secure to use them and
provoke spurious interrupts.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 3561a460f79d373faeda1948415e591eb8b511a1 dts: stm32: set the GPIOs used for wake-up as secure on stm32mp135f-dk]
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Change-Id: I872dae6b2e7e94de9eb06bc670cf5bf446a147eb
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/363846
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Etienne CARRIERE <etienne.carriere@foss.st.com>
---
 core/arch/arm/dts/stm32mp135f-dk.dts | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp135f-dk.dts b/core/arch/arm/dts/stm32mp135f-dk.dts
index 458a42f10..3cfa364a2 100644
--- a/core/arch/arm/dts/stm32mp135f-dk.dts
+++ b/core/arch/arm/dts/stm32mp135f-dk.dts
@@ -167,6 +167,14 @@
 	st,protreg = < (TZPROT(15)) >;
 };
 
+&gpiof {
+	st,protreg = < (TZPROT(8)) >;
+};
+
+&gpioi {
+	st,protreg = < (TZPROT(2) | TZPROT(3)) >;
+};
+
 &hash {
 	status = "okay";
 };
-- 
2.47.2

